### PR TITLE
add XOF currency

### LINKED ISSUE #5777: XOF Currency
Mali uses XOF (West African CFA franc). 

Symbol: CFA
Precision: 0
Rest same as euro

### PR DESCRIPTION
**Pull request by zachariah-at-msupply**
Fixes #5777

# 👩🏻‍💻 What does this PR do?

Adds XOF, XAF and CDF currencies.

## 💌 Any notes for the reviewer?

Believe in yourself 

;)

# 🧪 Testing

- [ ] In 4D msupply navigator go to special>stores>select the store you want>preferences>tick "Store: able to issue in foreign currency".
- [ ] Log in to that store in open mSupply and go to distribution>outbound shipment.
- [ ] Create a new outbound shipment to PF0-15 (or any other thing that's not a store).
- [ ] Add an item to that shipment, of something that you have some stock of, and put a pack qty issued in that column. Ideally, add an item that has a a price with two digits afteer the decimal place: e.g. $2.55 or similar (that way you can test the precision of the rounding)
- [ ] Click ok to get back to the page you can add medecines from, and click more in the top right corner.
- [ ] Scroll down the side bar that opens up and edit the 'Foreign Currency' field.
- [ ] Change the currency to XOF, and observe that
        1. the unit is XOF
        2. the price is rounded to 0 dp
- [ ] Change the currency to X**A**F, and observe that
        1. the unit is X**A**F
        2. the price is rounded to 0 dp
- [ ] Change the currency to CDF, and observe that
        1. the unit is CF
        2. the price is rounded to 2 dp
        
# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ x ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  4.

### CODE REVIEW

File: client/packages/common/src/intl/currency/currency.ts
Comment (roxy-dao) [inline client/packages/common/src/intl/currency/currency.ts:140]: Sorry the symbol is XOF

**Review by roxy-dao — CHANGES_REQUESTED** [review]
Thanks for the changes Zachariah! 

There were a few extra requests in the issue: 

Add XAF and CDF, the symbol is the code. and no precision (:

**Comment by zachariah-at-msupply** [comment]
I didn't see Mark's comments until now, I'll add XAF and CDF as he asked in the original issue and commit again.

**Comment by roxy-dao** [comment]
> I didn't see Mark's comments until now, I'll add XAF and CDF as he asked in the original issue and commit again before concluding the PR

You have to wait for approval before merging...

**Pull request merged**
